packages/ui/src/components/vuestic-mixins/ClickOutsideMixin/ClickOutsideMixin-utils.ts

```diff
@@ -0,0 +1,75 @@
+export const clickedRootScrollbar = (event: MouseEvent) => {
+  return (
+    document.documentElement.clientWidth < event.clientX ||
+    document.documentElement.clientHeight < event.clientY
+  )
+}
+
+export const ownerDocument = (node?: Element) => {
+  return (node && node.ownerDocument) || document
+}
+
+export const isLeftClickEvent = (event: MouseEvent) => {
+  return event.button === 0
+}
+
+export const isModifiedEvent = (event: MouseEvent) => {
+  return event.metaKey || event.altKey || event.ctrlKey || event.shiftKey
+}
+
+// HTML DOM and SVG DOM may have different support levels,
+// so we need to check on context instead of a document root element.
+export const contains = (context: Element, node: Element) => {
+  if (context.contains) {
+    return context.contains(node)
+  }
+
+  if (context.compareDocumentPosition) {
+    return context === node || !!(context.compareDocumentPosition(node) & 16)
+  }
+
+  return false
+}
+
+const __DEV__ = process.env.NODE_ENV !== 'production'
+
+export const handleMouseCapture = (e: MouseEvent, target: Element): boolean => {
```

Comment:
Is this solution taken from somewhere? Link?

Reply:
It's a mix from:
  1/ https://github.com/react-bootstrap/react-overlays/blob/master/src/useRootClose.ts
  2/ https://github.com/mui-org/material-ui/blob/master/packages/material-ui/src/ClickAwayListener/ClickAwayListener.js
Both links are in the head of the file